src/Talis/Persona/Client/OAuthClients.php

```diff
@@ -0,0 +1,154 @@
+<?php
+namespace Talis\Persona\Client;
+
+class OAuthClients extends Base
+{
+    /**
+     * Return an outh client
+     * @param string $clientId
+     * @param string $token
+     * @return array
+     * @throws \Exception
+     */
+    public function getOAuthClient($clientId, $token)
+    {
+        if(!is_string($clientId) || trim($clientId) === '')
+        {
+            throw new \InvalidArgumentException("Invalid clientId");
+        }
+        if(!is_string($token) || trim($token) === '')
+        {
+            throw new \InvalidArgumentException("Invalid token");
+        }
+
+        $url = $this->config['persona_host'].'/clients/'.$clientId;
+
+        try
+        {
+            $client = $this->personaGetOAuthClient($url, $token);
+            return $client;
+        } catch(\Exception $e)
+        {
+            throw new \Exception('OAuth client not found');
```

Comment:
what are the exceptions that can be caught here? IS an exception ONLY thrown if no oauth client is found?  are we trying to shield the callers of this code from what went wrong?